packages/toolpad-utils/src/collections.ts

```diff
@@ -93,3 +93,65 @@ export function filterKeys<U>(
 ): Record<string, U> {
   return mapProperties(obj, ([key, value]) => (filter(key) ? [key, value] : null));
 }
+
+/**
+ * check if two objects are deeply equal and support  Set, Map, Date, Array, Object
+ */
+// Reference code https://github.com/vuejs/vue/blob/49b6bd4264c25ea41408f066a1835f38bf6fe9f1/src/shared/util.ts#L297
+export function isDeepEqual<T = unknown>(obj: T, objToCompare: T): boolean {
```

Comment:
Originally I thought that this function would be expected to accept object types only. But because it seems that it can compare any type of values, let's probably rename `obj` and `objToCompare` to `value` and `valueToCompare`.